# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webcodecs/VideoDecoder.h
# Commit: cd30b650aeed
# Full Hash: cd30b650aeed44d1fb256a8aed378d27a3bfe39a
# Author: Chun-Min Chang <chun.m.chang@gmail.com>
# Date: 2023-06-29 15:31:45
# Regressor Bug: 1831451
# File Overlap Count: 1
# Description:
#   Bug 1831451 - Generate a unique Id for both ConfigureMessage and DecoderAgent r=alwu
#   
#   To batter tracking the ControlMessage logs, ConfigureMessage should
#   generate a unique Id for itself and its following DecodeMessages and
#   FlushMessages. Besides, that unique id can be assigned to DecoderAgent
# ==============================================================================

diff -r 6892b88cb4ee -r cd30b650aeed dom/media/webcodecs/VideoDecoder.h
--- a/dom/media/webcodecs/VideoDecoder.h	Thu Jun 29 04:15:04 2023 +0000
+++ b/dom/media/webcodecs/VideoDecoder.h	Thu Jun 29 04:15:04 2023 +0000
@@ -154,7 +154,8 @@
       UniquePtr<ControlMessage>& aMessage);
 
   // Returns true when mAgent can be created.
-  bool CreateDecoderAgent(UniquePtr<VideoDecoderConfig>&& aConfig,
+  bool CreateDecoderAgent(DecoderAgent::Id aId,
+                          UniquePtr<VideoDecoderConfig>&& aConfig,
                           UniquePtr<TrackInfo>&& aInfo);
   void DestroyDecoderAgentIfAny();
 
@@ -177,6 +178,9 @@
   uint32_t mDecodeQueueSize;
   bool mDequeueEventScheduled;
 
+  // A unique id tracking the ConfigureMessage and will be used as the
+  // DecoderAgent's Id.
+  uint32_t mLatestConfigureId;
   // Tracking how many decode data has been enqueued and this number will be
   // used as the DecodeMessage's Id.
   size_t mDecodeCounter;
